dev/com.ibm.ws.concurrent.mp_fat/test-applications/MPConcurrentCDIApp/src/concurrent/mp/fat/cdi/web/MPConcurrentCDITestServlet.java

```diff
@@ -0,0 +1,222 @@
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.inject.Inject;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.servlet.annotation.WebServlet;
+
+import org.eclipse.microprofile.concurrent.ManagedExecutor;
+import org.eclipse.microprofile.concurrent.ManagedExecutorConfig;
+import org.eclipse.microprofile.concurrent.ThreadContext;
+import org.junit.Test;
+
+import componenttest.app.FATServlet;
+
+@SuppressWarnings("serial")
+@WebServlet(urlPatterns = "/MPConcurrentCDITestServlet")
+public class MPConcurrentCDITestServlet extends FATServlet {
+
+    static final int TIMEOUT_MIN = 2;
+
+    @Inject
+    ConcurrencyBean bean;
+
+    @Inject
+    ManagedExecutor noAnno;
+
+    @Inject
+    @ManagedExecutorConfig
+    ManagedExecutor defaultAnno;
+
+    @Inject
+    @ManagedExecutorConfig(maxAsync = -1, maxQueued = -1, propagated = ThreadContext.ALL_REMAINING, cleared = ThreadContext.TRANSACTION)
+    ManagedExecutor defaultAnnoVerbose;
+
+    @Inject
+    @ManagedExecutorConfig(maxAsync = 5)
+    ManagedExecutor maxAsync5;
+
+    @Inject
+    @ManagedExecutorConfig(maxAsync = 2, maxQueued = 2)
+    ManagedExecutor max2;
+
+    @Inject
+    @ManagedExecutorConfig(propagated = {}, cleared = { ThreadContext.APPLICATION })
+    ManagedExecutor noAppCtx;
+
+    @Inject
+    @ManagedExecutorConfig(propagated = { ThreadContext.APPLICATION, ThreadContext.TRANSACTION }, cleared = {})
+    ManagedExecutor propogatedAB;
+
+    @Inject
+    @ManagedExecutorConfig(propagated = { ThreadContext.TRANSACTION, ThreadContext.APPLICATION }, cleared = {})
+    ManagedExecutor propogatedBA;
+
+    @Test
+    public void testMEDefaultsEqual() {
+        assertEquals(noAnno, defaultAnno);
+        assertEquals(noAnno, bean.getNoAnno());
+        assertEquals(noAnno, defaultAnno);
+        assertEquals(noAnno, bean.getDefaultAnno());
+        assertEquals(noAnno, defaultAnnoVerbose);
+    }
+
+    @Test
+    public void testMEConfiguredEqual() {
+        assertEquals(maxAsync5, bean.getMaxAsync5());
+    }
+
+    @Test
+    public void testMEDifferent() {
+        assertNotSame(noAnno, maxAsync5);
+        assertNotSame(noAnno, propogatedAB);
+    }
+
+    @Test
+    public void testMEPropogationEqual() {
+        assertEquals(propogatedAB, propogatedBA);
+    }
+
+    @Test
+    public void testBasicMEWorks() throws Exception {
+        CompletableFuture<String> cf = noAnno.supplyAsync(() -> {
+            try {
+                System.out.println("testBasicMEWorks: Performing lookup of 'foo'");
+                // TODO: This confirms JEE context is on the thread, need to find an operation that
+                // confirms 'application' context is on the thread
```

Comment:
Application context also includes thread context class loader, so that could be used as well.  If we do that, JNDI lookups could be dropped and we could disable the jndi-1.0 feature, which would be nice because the other test server has it enabled.